classes/plugins/DOIPubIdExportPlugin.inc.php

```diff
@@ -100,16 +105,166 @@ function manage($args, $request) {
 	}
 
 	/**
+	 * @copydoc ImportExportPlugin::display()
+	 */
+	function display($args, $request) {
+		parent::display($args, $request);
+
+		$context = $request->getContext();
+		switch (array_shift($args)) {
+			case 'index':
+			case '':
+				// Check for configuration errors:
+				$configurationErrors = array();
+				// 1) missing DOI prefix
+				$doiPrefix = $exportArticles = $exportIssues = null;
+				$pubIdPlugins = PluginRegistry::loadCategory('pubIds', true);
+				if (isset($pubIdPlugins['doipubidplugin'])) {
+					$doiPlugin = $pubIdPlugins['doipubidplugin'];
+					$doiPrefix = $doiPlugin->getSetting($context->getId(), $doiPlugin->getPrefixFieldName());
+					$exportArticles = $doiPlugin->getSetting($context->getId(), 'enableSubmissionDoi');
+					$exportIssues = $doiPlugin->getSetting($context->getId(), 'enableIssueDoi');
+					$exportRepresentations = $doiPlugin->getSetting($context->getId(), 'enableRepresentationDoi');
+				}
+				if (empty($doiPrefix)) {
+					$configurationErrors[] = DOI_EXPORT_CONFIG_ERROR_DOIPREFIX;
+				}
+
+				// 2) missing plugin settings
+				$form = $this->_instantiateSettingsForm($context);
+				foreach($form->getFormFields() as $fieldName => $fieldType) {
+					if ($form->isOptional($fieldName)) continue;
+					$pluginSetting = $this->getSetting($context->getId(), $fieldName);
+					if (empty($pluginSetting)) {
+						$configurationErrors[] = DOI_EXPORT_CONFIG_ERROR_SETTINGS;
+						break;
+					}
+				}
+				// Add link actions
+				$actions = $this->getExportActions($context);
+				$actionNames = array_intersect_key($this->getExportActionNames(), array_flip($actions));
+				import('lib.pkp.classes.linkAction.request.NullAction');
+				$linkActions = array();
+				foreach ($actionNames as $action => $actionName) {
+					$linkActions[] = new LinkAction($action, new NullAction(), $actionName);
+				}
+				$templateMgr = TemplateManager::getManager($request);
+				$templateMgr->assign('plugin', $this);
+				$templateMgr->assign('linkActions', $linkActions);
+				$templateMgr->assign('configurationErrors', $configurationErrors);
+				$templateMgr->assign('exportArticles', $exportArticles);
+				$templateMgr->assign('exportIssues', $exportIssues);
+				$templateMgr->assign('exportRepresentations', $exportRepresentations);
+				$templateMgr->display($this->getTemplatePath() . 'index.tpl');
+				break;
+			case 'exportSubmissions':
+			case 'exportIssues':
+				$selectedSubmissions = (array) $request->getUserVar('selectedSubmissions');
+				$selectedIssues = (array) $request->getUserVar('selectedIssues');
+
+				if (empty($selectedSubmissions) && empty($selectedIssues)) {
+					echo __('plugins.importexport.common.error.noObjectsSelected');
```

Comment:
If this is dumping an error message to the screen, maybe better to use a template, or `fatalError` if it's not something that will frequently be presented to the user.